bootstrap/012_task_log.sql

```diff
@@ -0,0 +1,53 @@
+
+create or replace function internal.task_run_id()
+returns text
+AS
+$$
+     SYSTEM$TASK_RUNTIME_INFO('CURRENT_TASK_GRAPH_RUN_GROUP_ID')
+$$;
+
+-- Generic table that tasks should record their execution into.
+CREATE TABLE INTERNAL.TASK_LOG IF NOT EXISTS (task_start timestamp_ltz, success boolean, task_name varchar, object_name varchar,
+    input variant, output variant, task_finish timestamp_ltz, task_run_id text, query_id text);
+
+create or replace procedure internal.migrate_task_logs()
+    returns text
+    language sql
+AS
+BEGIN
+    -- If the old task log tables exist, copy the rows into the new task_log table
+    if (exists (select * from information_schema.tables where table_schema = 'INTERNAL' and table_name = 'TASK_QUERY_HISTORY')) then
+        INSERT INTO INTERNAL.TASK_LOG (task_start, success, task_name, object_name, input, output, task_finish)
```

Comment:
can you put these ( and the below) into a transaction? The insert and teh trunat. Just to ensure they happen atomically

Reply:
Ugh, now Github shows me your comment.

I ended up switching the truncate table .. to a drop table (we don't need to keep these empty tables laying around). I think your ask is then moot as the DDL is going to auto-commit. LMK if I missed something.

Reply:
changed again after Jacques' feedback to not be destructive over the old task log tables.

This sets a value in internal.config to indicate we've migrated the task logs to the new table.